File: Programmers/Segun/2주차(Brute_force)/최소직사각형.cpp
//주어진 가로,세로 길이중 작은 값들중에 최댓값을 구하고
//주어진 가로, 세로 길이중 큰 값들중에 최댓값을 구해서 곱해준다.

#include <string>
#include <vector>
#include<algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int max1=0,max2=0;
    for(int i=0;i<sizes.size();i++){
        max1=max(max1,min(sizes[i][1],sizes[i][0]));    //가로 세로 중 작은 값들중의 최댓값
        max2=max(max2, max(sizes[i][1],sizes[i][0]));   //가로 세로 중 큰 값들중의 최댓값
    }
    return max1*max2;
}

File: Programmers/BoHwa/2주차(Brute_Force)/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int max1 = 0;
        int max2 = 0; //초기화 꼭 하기!

        for (int i = 0; i < sizes.length; i++) {
            for (int j = 0; j < sizes[i].length; j++) {
                if (max1 < sizes[i][j]) { //가장 큰 값을 max1에 저장
                    max1 = sizes[i][j];
                }
            }
        }

        for (int i = 0; i < sizes.length; i++) {
            if (max2 < sizes[i][0] && max2 < sizes[i][1]) {//max2보다 비교하는 배열의 값이 모두 크다면
                max2 = Math.min(sizes[i][0], sizes[i][1]);//배열의 둘 중 작은 값을 max2에 대입
            }
        }

        answer=max1*max2;

        return answer;
    }
}
